pkgs/by-name/ze/zed-editor/package.nix

```
@@ -0,0 +1,220 @@
+      "tree-sitter-nu-0.0.1" = "sha256-V6EZfba5e0NdOG4n3DNI25luNXfcCN3+/vNYuz9llUk=";
+      "tree-sitter-ocaml-0.20.4" = "sha256-ycmjIKfrsVSVHmPP3HCxfk5wcBIF/JFH8OnU8mY1Cc8=";
+      "tree-sitter-proto-0.0.2" = "sha256-W0diP2ByAXYrc7Mu/sbqST6lgVIyHeSBmH7/y/X3NhU=";
+      "tree-sitter-racket-0.0.1" = "sha256-ie64no94TtAWsSYaBXmic4oyRAA01fMl97+JWcFU1E8=";
+      "tree-sitter-scheme-0.2.0" = "sha256-K3+zmykjq2DpCnk17Ko9LOyGQTBZb1/dgVXIVynCYd4=";
+      "tree-sitter-toml-0.5.1" = "sha256-5nLNBxFeOGE+gzbwpcrTVnuL1jLUA0ZLBVw2QrOLsDQ=";
+      "tree-sitter-typescript-0.20.2" = "sha256-cpOAtfvlffS57BrXaoa2xa9NUYw0AsHxVI8PrcpgZCQ=";
+      "tree-sitter-vue-0.0.1" = "sha256-8v2e03A/Uj6zCJTH4j6TPwDQcNFeze1jepMADT6UVis=";
+      "tree-sitter-yaml-0.0.1" = "sha256-S59jLlipBI2kwFuZDMmpv0TOZpGyXpbAizN3yC6wJ5I=";
+      "tree-sitter-zig-0.0.1" = "sha256-a3W7eBUN4V3HD3YPr1+3tpuWQfIQy1Wu8qxCQx0hEnI=";
+    };
+  };
+
+  nativeBuildInputs = [
+    curl
+    pkg-config
+    protobuf
+    rustPlatform.bindgenHook
+  ] ++ lib.optionals stdenv.isDarwin [ xcbuild.xcrun ];
+
+  buildInputs =
+    [
+      curl
+      fontconfig
+      freetype
+      libgit2
+      openssl
+      sqlite
+      zlib
+      zstd
+    ]
+    ++ lib.optionals stdenv.isLinux [
+      alsa-lib
+      libxkbcommon
+      wayland
+      xorg.libxcb
+    ]
+    ++ lib.optionals stdenv.isDarwin (
+      with darwin.apple_sdk.frameworks;
+      [
+        AppKit
+        CoreAudio
+        CoreFoundation
+        CoreGraphics
+        CoreMedia
+        CoreServices
+        CoreText
+        Foundation
+        IOKit
+        Metal
+        Security
+        SystemConfiguration
+        VideoToolbox
+      ]
+    );
+
+  buildFeatures = [ "gpui/runtime_shaders" ];
+
+  env = {
+    ZSTD_SYS_USE_PKG_CONFIG = true;
+    FONTCONFIG_FILE = makeFontsConf {
+      fontDirectories = [
+        "${src}/assets/fonts/zed-mono"
+        "${src}/assets/fonts/zed-sans"
+      ];
+    };
+  };
+
+  postFixup = lib.optionalString stdenv.isLinux ''
+    patchelf --add-rpath ${vulkan-loader}/lib $out/bin/*
+    patchelf --add-rpath ${wayland}/lib $out/bin/*
+  '';
+
+  # Fail with "unimplemented error"
+  checkFlags = lib.optionals stdenv.hostPlatform.isLinux [
+    "--skip=test_base_keymap"
+    "--skip=test_bundled_settings_and_themes"
+    "--skip=test_disabled_keymap_binding"
+    "--skip=test_navigation"
+    "--skip=test_new_empty_workspace"
+    "--skip=test_open_and_save_new_file"
+    "--skip=test_open_entry"
+    "--skip=test_opening_excluded_paths"
+    "--skip=test_open_paths"
+    "--skip=test_open_paths_action"
+    "--skip=test_pane_actions"
+    "--skip=test_reopening_closed_items"
+    "--skip=test_save_conflicting_item"
+    "--skip=test_setting_language_when_saving_as_single_file_worktree"
+    "--skip=test_window_edit_state"
+  ];
+
+  postInstall = ''
+    mv $out/bin/Zed $out/bin/zed-stable
+    ln -s $out/bin/zed-stable $out/bin/zed
```

```suggestion
    mv $out/bin/Zed $out/bin/zed
```
This doesn't seem to be an upstream convention to do this.

---

It's an approximation of the (WIP) upstream Linux packaging script: https://github.com/zed-industries/zed/blob/main/script/bundle-linux#L95

---

As mentioned in the comments above `zed` should really be a link to the cli, but `buildRustPackage` has no way of specifying multiple targets.